src/context/GlobalState.tsx

```diff
@@ -1,28 +1,30 @@
+import { getDocs } from "firebase/firestore";
 import React, { createContext, useReducer, useEffect } from "react";
+import { colref, db } from "../firebase";
 import { ACTION } from "./Actions";
 import AppReducer from "./Reducer";
 
-const url = "https://fakestoreapi.com/products";
+// const url = "https://fakestoreapi.com/products";
 
 const initialState: any = {
   data: [],
   cart: [],
+  details: {},
+  res: [],
 };
```

Comment:
- We should avoid using `any` for state and have static types in place.
- What does `res` holds is unclear. If it's meant for products, it should be named `products` and should have a specific data type associated with it.